ouroboros-network/test/CDDL.hs

```diff
@@ -149,7 +150,7 @@ unpackResult :: IO (ExitCode, ByteString, ByteString) -> IO ByteString
 unpackResult r = r >>= \case
     (ExitFailure _, _, err) -> error $ Char8.unpack err
     (ExitSuccess, bytes, err) -> do
-            Char8.putStrLn err -- dump debugging output
+            Char8.hPutStr stderr err
```

Comment:
Why do you need this output written to `stderr`? It ran successfully, why would we want to see anything other than quickcheck result?

Reply:
The script calls the external cddl-tool
and that tool sometimes prints out some warnings to stderr,
For example if the spec contains unreachable declarations.
Sometimes I comment-out parts in the cddl-spec which makes other parts unreachable
and triggers that warnings and I don't want to forget to turn it on again.
In the code I could
1) ignore the warnings alltogether
2) forward the warnings
3) make every warning an error.

Reply:
I'd prefer making the warnings an error so that the test suite isn't dependent on manual inspection.
The suite may need `cddl-tool` but being able to run it in CI would be very useful (as shown by the list encoding bug you found).

Reply:
@karknu
OK done that.
BTW the tests had been running in CI for some while,
only they are not mandatory-(successful) CI tests.
They don't show up in PRs ob Github but on Hydra.